ts/progress-bar/progress-bar.service.ts

```diff
@@ -0,0 +1,126 @@
+module Lui.Service {
+	"use strict";
+
+//==========================================
+// ---- inspired by https://github.com/VictorBjelkholm/ngProgress/blob/master/src/provider.js
+// ==========================================
+	export class ProgressBarService {
+		public static IID: string = "progressBarService";
+		public static $inject: string[] = ["$document", "$window", "$rootScope", "$timeout", "$interval"];
+		public latencyThreshold = 200;
+		private httpResquestListening: boolean = false;
+		private $document: angular.IDocumentService;
+		private $window: angular.IWindowService;
+		private $rootScope: ng.IScope;
+		private $timeout: ng.ITimeoutService;
+		private $interval: ng.IIntervalService;
+		private status: number = 0;
+		private currentPromiseInterval: ng.IPromise<any>;
+		private completeTimeout: ng.IPromise<any>;
+		private parent: angular.IAugmentedJQuery = angular.element(document).find("body");
+		private progressBarTemplate: string = '<div class="lui slim secondary progressing progress progress-bar"><div class="indicator" data-percentage="0" style="width: 0%;"></div></div>';
+		private progressbarEl: angular.IAugmentedJQuery;
+		private isStarted: boolean;
+
+		constructor(
+			$document: angular.IDocumentService,
+			$window: angular.IWindowService,
+			$rootScope: ng.IScope,
+			$timeout: ng.ITimeoutService,
+			$interval: ng.IIntervalService) {
+			this.$document = $document;
+			this.$window = $window;
+			this.$rootScope = $rootScope;
+			this.$timeout = $timeout;
+			this.$interval = $interval;
+		}
+
+		public addProgressBar = (parent?: angular.IAugmentedJQuery) => {
+			if (!parent) {
+				parent = this.parent;
+			}
+			if (!!this.progressbarEl) {
+				this.progressbarEl.remove();
+			}
+			this.progressbarEl = angular.element(this.progressBarTemplate);
+			parent.append(this.progressbarEl);
+		};
+
+		public setColor = (color: string) => {
+			this.progressbarEl.children().css("background", color);
+		};
```

Comment:
I dont like that, at all. you should not specify the color of the progress bar by giving it a color or a hex but you you should give it a pallete

i'd rather have and no method `setColor`

```ts
public addProgressBar = (parent?: angular.IAugmentedJQuery, palette? = "primary") => {
	...
};
```